packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -933,13 +937,16 @@ Widget _buildCupertinoDialogTransitions(BuildContext context, Animation<double>
 /// It is only used when the method is called. Its corresponding widget can
 /// be safely removed from the tree before the dialog is closed.
 ///
+/// The `useRootNavigator` argument is used to determine whether to push the
+/// dialog to the [Navigator] furthest from or nearest to the given `context`.
+///
 /// Returns a [Future] that resolves to the value (if any) that was passed to
 /// [Navigator.pop] when the dialog was closed.
 ///
-/// The dialog route created by this method is pushed to the root navigator.
-/// If the application has multiple [Navigator] objects, it may be necessary to
-/// call `Navigator.of(context, rootNavigator: true).pop(result)` to close the
-/// dialog rather than just `Navigator.pop(context, result)`.
+/// By default, the dialog route created by this method is pushed to the root
```

Comment:
This paragraph should probably come right after the paragraph that explains "useRootNavigator"